pkg/proc/native/proc.go

```diff
@@ -370,29 +370,38 @@ func (dbp *nativeProcess) writeSoftwareBreakpoint(thread *nativeThread, addr uin
 	return err
 }
 
-func openRedirects(redirects [3]string, foreground bool) (stdin, stdout, stderr *os.File, closefn func(), err error) {
-	toclose := []*os.File{}
+func openRedirects(redirects [3]proc.OutputRedirect, foreground bool) (stdin, stdout, stderr *os.File, closefn func(), err error) {
+	var (
+		toclose = []*os.File{}
+	)
 
-	if redirects[0] != "" {
-		stdin, err = os.Open(redirects[0])
+	if redirects[0].Path != "" {
+		stdin, err = os.Open(redirects[0].Path)
 		if err != nil {
 			return nil, nil, nil, nil, err
 		}
 		toclose = append(toclose, stdin)
-	} else if foreground {
+	} else {
```

Comment:
Same as the other, can you explain this change? This in particular just looks wrong. We only want to pass stdin to the target process if we are foregrounding it.

Reply:
done.